signatures/pdf_sign_new_page.go

```diff
@@ -0,0 +1,215 @@
+ */
+package main
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/unidoc/unipdf/v3/annotator"
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/core"
+	"github.com/unidoc/unipdf/v3/model"
+	"github.com/unidoc/unipdf/v3/model/sighandler"
+)
+
+const licenseKey = `
+-----BEGIN UNIDOC LICENSE KEY-----
+Free trial license keys are available at: https://unidoc.io/
+-----END UNIDOC LICENSE KEY-----
+`
+
+func init() {
+	// Enable debug-level logging.
+	// unicommon.SetLogger(unicommon.NewConsoleLogger(unicommon.LogLevelDebug))
+
+	err := license.SetLicenseKey(licenseKey, `Company Name`)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func main() {
+	args := os.Args
+	if len(args) < 3 {
+		log.Fatalln("Usage: go run pdf_sign_new_page INPUT_PDF_PATH OUTPUT_PDF_PATH")
+	}
+
+	inputPath := args[1]
+	outputPath := args[2]
+
+	// Read the original file.
+	f, err := os.Open(inputPath)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+	defer f.Close()
+
+	pdfReader, err := model.NewPdfReader(f)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+
+	// Add new page and write it into a temporary file.
+	tempPath, err := addPage(pdfReader)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+
+	// Reread the temporary file.
+	ftmp, err := os.Open(tempPath)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+	defer f.Close()
+
+	pdfReader, err = model.NewPdfReader(ftmp)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+
+	totalPage, err := pdfReader.GetNumPages()
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+
+	// Add signature and write it to output.pdf file.
+	err = addSignature(pdfReader, totalPage, outputPath)
+	if err != nil {
+		log.Fatalf("Fail: %v\n", err)
+	}
+
+	fmt.Println("Done")
+}
+
+func addPage(reader *model.PdfReader) (temporaryFilePath string, err error) {
+	tmpWriter, err := reader.ToWriter(&model.ReaderToWriterOpts{})
+	if err != nil {
+		return "", err
+	}
+
+	err = tmpWriter.AddPage(model.NewPdfPage())
+	if err != nil {
+		return "", err
+	}
+
+	tempPath := filepath.Join(os.TempDir(), "output.pdf")
+
+	err = tmpWriter.WriteToFile(tempPath)
+	if err != nil {
+		return "", err
+	}
+
+	return tempPath, nil
+}
+
+func addSignature(reader *model.PdfReader, pageNum int, outputPath string) error {
+	// Create appender.
+	appender, err := model.NewPdfAppender(reader)
+	if err != nil {
+		return err
+	}
+
+	// Generate key pair.
+	priv, cert, err := generateSigKeys()
+	if err != nil {
+		return err
+	}
+
+	// Create signature handler.
+	handler, err := sighandler.NewAdobePKCS7Detached(priv, cert)
+	if err != nil {
+		return err
+	}
+
+	// Create signature.
+	signature := model.NewPdfSignature(handler)
+	signature.SetName("Test Signature Appearance Name")
+	signature.SetReason("TestSignatureAppearance Reason")
+	signature.SetDate(time.Now(), "")
```

Comment:
You don't seem to be initializing the signature:
```go
if err := signature.Initialize(); err != nil {
    return err
}
```

If you omit this step, the output file will have the signature field, but the file will not be signed.